src/engine/source/router/include/router/environmentManager.hpp

```diff
@@ -1,77 +1,66 @@
-    using concurrentQueue = moodycamel::BlockingConcurrentQueue<std::string>;
-
-    std::shared_ptr<builder::Builder> m_builder; ///< Builder for environment creation
-    std::shared_ptr<concurrentQueue> m_queue;    ///< Queue for environment
-    std::size_t m_numThreads; ///< Number of threads for each environment
-
-    /**
-     * @brief Map of active environments
-     */
-    std::unordered_map<std::string, activeEnvironment> m_environments;
+    /* Status */
+    std::unordered_map<std::string, std::vector<RuntimeEnvironment>> m_environments; ///< Map of environments
     std::shared_mutex m_mutex; ///< Mutex to protect the environments map
 
-    /**
-     * @brief API callback for environment creation
-     * @param params Parameters for environment creation ("/name")
-     * @return api::WazuhResponse with the result of the operation
-     */
-    api::WazuhResponse apiSetEnvironment(const json::Json& params);
-
-    /**
-     * @brief API callback for environment status
-     *
-     * @param params Parameters for environment status ("/name")
-     * @return api::WazuhResponse with an array of environments
-     */
-    api::WazuhResponse apiGetEnvironment(const json::Json& params);
+    /* Config */
+    const std::size_t m_numInstances; ///< Number of instances of each environment
 
-    /**
-     * @brief API callback for environment deletion
-     *
-     * @param params Parameters for environment deletion ("/name")
-     * @return api::WazuhResponse with the result of the operation
-     */
-    api::WazuhResponse apiDelEnvironment(const json::Json& params);
+    /* Resources */
+    std::shared_ptr<builder::Builder> m_builder; ///< Builder for environment creation
 
 public:
     /**
-     * @brief Construct a new Environment Manager object
+     * @brief Create the environment manager
      *
+     * The environment manager is responsible to manage the runtime environment, creation, destruction, and interaction
+     * with the environment
      * @param builder Builder for environment creation
-     * @param queue Queue for environment
-     * @param numThreads Number of threads for each enviroment
+     * @param maxInstances Number of instances of each environment
      */
-    EnvironmentManager(std::shared_ptr<builder::Builder> builder,
-                       std::shared_ptr<concurrentQueue> queue,
-                       std::size_t numThreads)
-        : m_builder(builder)
-        , m_queue(queue)
-        , m_numThreads(numThreads)
-        , m_environments()
-        , m_mutex() {};
+    EnvironmentManager(std::shared_ptr<builder::Builder> builder, std::size_t maxInstances)
+        : m_environments {}
+        , m_mutex {}
+        , m_numInstances {maxInstances}
+        , m_builder {builder}
+    {
+        if (maxInstances == 0)
+        {
+            throw std::runtime_error("EnvironmentManager: Number of instances of the environment can't be 0.");
```

Comment:
Please, let's try to avoid short expressions like "can't" in messages

Reply:
done